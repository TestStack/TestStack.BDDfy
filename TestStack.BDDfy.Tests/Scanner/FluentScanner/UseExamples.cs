using NUnit.Framework;

namespace TestStack.BDDfy.Tests.Scanner.FluentScanner
{
    class UseExamples
    {
        private int _start;
        private int _eat;

        [Test]
        public void RunExamplesWithFluentApi()
        {
            this.Given(_ => _.GivenThereAre__start__Cucumbers(Args.From<int>("Start")))
                .When(_ => _.WhenIEat__eat__Cucumbers(Args.From<int>("Eat")))
                .Then(_ => _.ThenIShouldHave__left__Cucumbers(Args.From<int>("Left")))
                .WithExamples(
                    new object[] { "Start", "Eat", "Left" },
                    new object[] { 12, 5, 8 },
                    new object[] { 20, 5, 17 })
                .BDDfy();
        }

        [Test]
        public void RunExamplesWithReflectiveApi()
        {
            this.WithExamples(
                    new object[] { "start", "eat", "left" },
                    new object[] { 12, 5, 7 },
                    new object[] { 20, 5, 15 })
                .BDDfy();
        }

        private void GivenThereAre__start__Cucumbers(int start)
        {
            _start = start;
        }

        [AndGiven("And I eat <eat> of them")]
        private void WhenIEat__eat__Cucumbers(int eat)
        {
            _eat = eat;
        }

        private void ThenIShouldHave__left__Cucumbers(int left)
        {
            Assert.That(_start - _eat, Is.EqualTo(left));
        }

        [Test]
        public void GettingTitle()
        {
            Assert.That(NetToString.Convert("GivenThereAre__start__Cucumbers"), Is.EqualTo("Given there are <start> cucumbers"));
            Assert.That(NetToString.Convert("Given_there_are__start__cucumbers"), Is.EqualTo("Given there are <start> cucumbers"));
        }
    }
}
