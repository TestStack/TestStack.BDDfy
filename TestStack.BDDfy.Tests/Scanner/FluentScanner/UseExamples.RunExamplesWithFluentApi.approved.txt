
Scenario: Run examples with fluent api
	Given there are <start> cucumbers  
	When I eat <eat> cucumbers         
	Then I should have <left> cucumbers

Examples: 
|        | Start | Eat | Left | Errors                                                                                                             |
| Passed | 12    | 5   | 7    |                                                                                                                    |
| Failed | 20    | 5   | 17   | Step: Then I should have <left> cucumbers failed with exception: [Expected: 17,  But was: 15] [Details at 1 below] |

Exceptions:
  1. Expected: 17,  But was: 15
   at NUnit.Framework.Assert.That(Object actual, IResolveConstraint expression, String message, Object[] args)
   at NUnit.Framework.Assert.That(Object actual, IResolveConstraint expression)
   at TestStack.BDDfy.Tests.Scanner.FluentScanner.UseExamples.ThenIShouldHave__left__Cucumbers(Int32 left) in c:\Users\Jake\_Code\TestStack.BDDfy\TestStack.BDDfy.Tests\Scanner\FluentScanner\UseExamples.cs:line 64
   at lambda_method(Closure , UseExamples )
   at TestStack.BDDfy.StepActionFactory.<>c__DisplayClass8`1.<>c__DisplayClassa.<GetStepAction>b__7() in c:\Users\Jake\_Code\TestStack.BDDfy\TestStack.BDDfy\Scanners\StepScanners\StepActionFactory.cs:line 22
   at TestStack.BDDfy.StepActionFactory.Run(Func`1 func) in c:\Users\Jake\_Code\TestStack.BDDfy\TestStack.BDDfy\Scanners\StepScanners\StepActionFactory.cs:line 40
   at TestStack.BDDfy.StepActionFactory.<>c__DisplayClass8`1.<GetStepAction>b__6(Object o) in c:\Users\Jake\_Code\TestStack.BDDfy\TestStack.BDDfy\Scanners\StepScanners\StepActionFactory.cs:line 20
   at TestStack.BDDfy.Step.Execute(Object testObject) in c:\Users\Jake\_Code\TestStack.BDDfy\TestStack.BDDfy\Step.cs:line 54
   at TestStack.BDDfy.Scenario.ExecuteStep(Step step) in c:\Users\Jake\_Code\TestStack.BDDfy\TestStack.BDDfy\Scenario.cs:line 58
   at TestStack.BDDfy.Processors.ExceptionProcessor.Process(Story story) in c:\Users\Jake\_Code\TestStack.BDDfy\TestStack.BDDfy\Processors\ExceptionProcessor.cs:line 97
   at TestStack.BDDfy.Engine.Run() in c:\Users\Jake\_Code\TestStack.BDDfy\TestStack.BDDfy\Engine.cs:line 43
   at TestStack.BDDfy.Tests.Scanner.FluentScanner.UseExamples.<>c__DisplayClass1.<RunExamplesWithFluentApi>b__0() in c:\Users\Jake\_Code\TestStack.BDDfy\TestStack.BDDfy.Tests\Scanner\FluentScanner\UseExamples.cs:line 34
   at NUnit.Framework.Assert.Throws(IResolveConstraint expression, TestDelegate code, String message, Object[] args)

